packaging_automation/citus_package.py

```diff
@@ -0,0 +1,264 @@
+            raise ValueError(f"{os_release} is not among supported releases for {os_name}."
+                             f"Supported releases are as below:\n {','.join(supported_platforms[os_name])}")
+    return os_name, os_release
+
+
+def is_docker_running() -> bool:
+    docker_client = docker.from_env()
+    try:
+        docker_client.ping()
+        return True
+    except:
+        return False
+
+
+@validate_parameters
+def get_signing_credentials(packaging_secret_key: str,
+                            packaging_passphrase: non_empty(non_blank(str))) -> Tuple[str, str]:
+    if packaging_secret_key:
+        secret_key = packaging_secret_key
+    else:
+        fingerprints = get_gpg_fingerprints_by_name(GPG_KEY_NAME)
+        if len(fingerprints) == 0:
+            raise ValueError(f"Key for {GPG_KEY_NAME} does not exist")
+
+        gpg = gnupg.GPG()
+
+        private_key = gpg.export_keys(fingerprints[0], secret=True, passphrase=packaging_passphrase)
+        secret_key = transform_key_into_base64_str(private_key)
+
+    passphrase = packaging_passphrase
+    return secret_key, passphrase
+
+
+def sign_packages(base_output_path: str, sub_folder: str, secret_key: str, passphrase: str, input_files_dir: str,
+                  output_validation: bool = False):
+    output_path = f"{base_output_path}/{sub_folder}"
+    deb_files = glob.glob(f"{output_path}/*.deb", recursive=True)
+    rpm_files = glob.glob(f"{output_path}/*.rpm", recursive=True)
+    os.environ["PACKAGING_PASSPHRASE"] = passphrase
+    os.environ["PACKAGING_SECRET_KEY"] = secret_key
+
+    if len(rpm_files) > 0:
+        print("Started RPM Signing...")
+        result = run_with_output(f"docker run --rm -v {output_path}:/packages/{sub_folder} -e PACKAGING_SECRET_KEY -e "
+                                 f"PACKAGING_PASSPHRASE citusdata/packaging:rpmsigner", text=True)
+
+        if result.returncode != 0:
+            raise ValueError(f"Error while signing rpm files.Err:{result.stderr}")
+        print("RPM signing finished successfully.")
+        output = result.stdout
+        print(output)
+
+        if output_validation:
+            validate_output(output, f"{input_files_dir}/packaging_ignore.yml", PackageType.rpm)
+
+    if len(deb_files) > 0:
+        print("Started DEB Signing...")
+
+        result = subprocess.run(
+            ["docker", "run", "--rm", "-v", f"{output_path}:/packages/{sub_folder}",
+             "-e", "PACKAGING_SECRET_KEY", "-e", "PACKAGING_PASSPHRASE", "citusdata/packaging:debsigner"],
+            text=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, input=passphrase)
+
+        print("Result" + result.stdout)
+
+        if result.returncode != 0:
+            raise ValueError(f"Error while signing DEB files.Err:{result.stdout}")
+
+        if output_validation:
+            validate_output(result.stdout, f"{input_files_dir}/packaging_ignore.yml", PackageType.deb)
+        print("DEB signing finished successfully.")
```

Comment:
Please make the order of print operations similar on deb and rpm packages. Right now, they are not really easy to follow.

RPM:
- check return code
- "RPM signing finished successfully."
- docker run output `print(output)`
- validation

DEB:
- docker run output `print("Result" + result.stdout)`
- check return code
- validation
- "DEB signing finished successfully."

I think it makes more sense to do these operations in the order below:
- print("Started ..
- print stdout
- if stderr OR non-zero return code, print stderr and raise error
- validate output
- print("Finished ..

Reply:
Fixed

Reply:
Added  two more small comments